node_editor/src/components/StateEditor.vue

```diff
@@ -40,31 +40,26 @@ const elements = computed<FlowElements>({
 			}
 
 			stateHash[key] = el
+		}
 
+		for (const [key, value] of Object.entries(states.value)) {
 			if (value.on) {
 				for (const [edgeKey, edgeValue] of Object.entries(value.on)) {
-					if (Array.isArray(edgeValue)) {
-						for (const edge of edgeValue) {
-							// TODO: handle typescript errors for both types of states
-							// eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
-							const edgeJson = edge.toJSON()
-							// eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
-							const target = edgeJson.target.toString()
-							stateElements.push({
-								id: `${key}-${edgeKey}`,
-								target: target,
-								source: key,
-								label: edgeKey,
-								animated: true,
-							})
-
-							// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-							hasInputs[target] = true
-						}
-					}
+					const target = edgeValue.target || edgeValue
```

Comment:
If the proxy array value.on has more than one edge, edgeValue will contain a proxyArray object where `target` can be accessed. Otherwise, edgeValue will be properly pulled as the value.

Reply:
Great comment, let's put it in the code